src/openms/source/CHEMISTRY/ISOTOPEDISTRIBUTION/CoarseIsotopePatternGenerator.cpp

```diff
@@ -179,6 +179,71 @@ namespace OpenMS
     return result;
   }
 
+  // static
+  IsotopeDistribution CoarseIsotopePatternGenerator::approximateFromPeptideWeight(double mass, int num_peaks)
+  {
+    IsotopeDistribution result;
+    result.resize(num_peaks);
+
+    // lambda * mass. Lambda is the parameter for Poisson distribution. Value (1/1800) taken from Bellew et al
+    double factor = mass / 1800.0;
+
+    // values of (m * lambda) ^ k
+    double curr_power = 1.0;
+
+    // values of k!
+    UInt curr_factorial = 1;
+
+    // for k=0, non-normalized value is always 1
+    result[0] = Peak1D(mass, 1.0);
+
+    float curr_intensity;
+    for (UInt k = 1; k < num_peaks; ++k)
+    {
+      curr_power *= factor;
+      curr_factorial *= k;
+      curr_intensity = curr_power / curr_factorial;
+      result[k] = Peak1D(mass + (k * OpenMS::Constants::NEUTRON_MASS_U),
```

Comment:
maybe write how mass is calculated in the function description

Reply:
Done